reconstruction/ecoli/dataclasses/process/external_pathway.py

```diff
@@ -0,0 +1,264 @@
+from wholecell.utils import units
+import numpy as np
+from wholecell.utils import build_ode
+import scipy
+import scipy.integrate
+import sympy as sp
+from wholecell.utils import data
+
+#TO DO: reactions catalyzed by several enzymes.
+
+# Alternative methods to try (in order of priority) when solving ODEs to the next time step
+IVP_METHODS = ['LSODA', 'BDF']
+COUNTS_UNITS = units.mmol
+VOLUME_UNITS = units.L
+TIME_UNITS = units.s
+MODEL_FLUX_UNITS = COUNTS_UNITS / VOLUME_UNITS / TIME_UNITS
+DCW_FLUX_UNITS = units.mmol / units.g / units.h
+
+class MetabolismExternalPathway(object):
+    def __init__(self, raw_data, sim_data):
+
+        if hasattr(raw_data, 'new_gene_data') and hasattr(getattr(raw_data.new_gene_data, dir(raw_data.new_gene_data)[-1]), 'metabolic_reactions_new'):
```

Comment:
Also, if `raw_data` could always have a `.new_gene_data` or `.new_genes_option` field, then all the code using it gets simpler and easier to think about.

If that doesn't fit, `getattr(obj, field, default)` to avoid the need to check `hasattr()` before calling `getattr()`, e.g. `getattr(getattr(raw_data, 'new_gene_data', None), 'metabolic_gene', ())`, but clearly `raw_data.new_gene_data.metabolic_gene` is easier to work with.

Reply:
I believe `raw_data.new_genes_option` always exists, `raw_data.new_genes_option="off"` by default. @ioanagherman I think after discussing with @ggsun, @1fish2, and @thalassemia it makes sense to always include the new gene "optional" files in all subdirectories, so that way we can worry less about existence. I started this in PR #1431. 

What this means for the metabolic reactions is that we would always have the `metabolic_reactions_new.tsv` file in the new gene data folders, and then if there were no reactions in the file we would initialize an empty data structure so that way when you loop over it (like in line 45 of this file), nothing would happen.